Zend/zend_language_parser.y

```diff
@@ -262,6 +270,21 @@ start:
 	top_statement_list	{ CG(ast) = $1; }
 ;
 
+SEMI_RESERVED:
+	  T_INCLUDE | T_INCLUDE_ONCE | T_EVAL | T_REQUIRE | T_REQUIRE_ONCE | T_LOGICAL_OR | T_LOGICAL_XOR | T_LOGICAL_AND
+	| T_INSTANCEOF | T_NEW | T_CLONE | T_EXIT | T_IF | T_ELSEIF | T_ELSE | T_ENDIF | T_ECHO | T_DO | T_WHILE | T_ENDWHILE
+	| T_FOR | T_ENDFOR | T_FOREACH | T_ENDFOREACH | T_DECLARE | T_ENDDECLARE | T_AS | T_TRY | T_CATCH | T_FINALLY
+	| T_THROW | T_USE | T_INSTEADOF | T_GLOBAL | T_VAR | T_UNSET | T_ISSET | T_EMPTY | T_CONTINUE | T_GOTO
+	| T_FUNCTION | T_CONST | T_RETURN | T_PRINT | T_YIELD | T_LIST | T_SWITCH | T_ENDSWITCH | T_CASE | T_DEFAULT | T_BREAK
+	| T_ARRAY | T_CALLABLE | T_EXTENDS | T_IMPLEMENTS | T_NAMESPACE | T_CLASS | T_TRAIT | T_INTERFACE
+	// | T_STATIC | T_ABSTRACT | T_FINAL | T_PRIVATE | T_PROTECTED | T_PUBLIC
+;
+
+identifier:
+		T_STRING { $$ = $1; }
+	| 	/* if */ SEMI_RESERVED { REWIND } /* and rematch as */ T_STRING { $$ = $3; }
```

Comment:
Looks like a great idea because as we don't waste time interacting with the lexer anymore. On the other side I was doing the ext tokenizer port based on this lexical feedback and now the lexer is completely unaware of context again.

I'll use this anyway and will try to find another solution for the tokenizer extension side. Thanks ^^